### PR TITLE
Adding GHA clang-format test

### PR DESCRIPTION
**Pull request by allevin**
This PR will add a Github Actions script that runs on PR creation or commit to a PR.  It will clone the repo on Github's test systems and then run clang-format v12.  It will set labels and add appropriate comments based upon testing.  It will also set/remove the AT: WIP flag to lockout Autotester testing if the format test fails.

### CODE REVIEW

File: .github/workflows/PR_clang-format_test.yml [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        script: |
+            github.issues.addLabels({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              labels: ['AT: CLANG-FORMAT PASS']
+            })
+      if: ${{ success() }}
+
+    - name: REMOVE WIP Label Results for Success
```
Comment (calewis) [inline .github/workflows/PR_clang-format_test.yml:132]: So I need to learn more about actions, but is this going to run regardless of whether or not a user sets the WIP flag? Because we don't want to accidentally remove a WIP flag if it was set by the submitter.

**Review by calewis — APPROVED** [review]
Can we check that one comment I have about not removing submitter set WIP, it's an edge case, but we should double check.  I'm going to approve because I think this is good and it worked on my first order test on my fork and we can make changes if/when issues arise.

File: .github/workflows/PR_clang-format_test.yml (hunk 1 above)
Comment (allevin) [reply .github/workflows/PR_clang-format_test.yml:132]: The AT: WIP label is one way.  Also the user can set [WIP] in the title of the PR to have the same effect.   That might be a better solution for PR's that are not wanted to be processed.

**Comment by feldergast** [comment]
A few questions before we merge this:

1 - When will this run?  Only at PR submission?  After a new commit is pushed?
2 - Will this do more than add WIP to the PR (i.e., is there a check to make sure this is successful that will be required before merging)?  Or, are we leaving in the clang-format test in the autotester to make do the actual check that will be needed before merging?
3 - Should this have run when this PR was submitted (I don't see any comments that it generated), or will it only run once it is committed into devel?

**Review by feldergast — CHANGES_REQUESTED** [review]
Just going to set this as request changes to prevent accidentally merging before we get a few questions answered.  See comments.

**Comment by allevin** [comment]
> A few questions before we merge this:
> 
> 1 - When will this run? Only at PR submission? After a new commit is pushed?
> 2 - Will this do more than add WIP to the PR (i.e., is there a check to make sure this is successful that will be required before merging)? Or, are we leaving in the clang-format test in the autotester to make do the actual check that will be needed before merging?
> 3 - Should this have run when this PR was submitted (I don't see any comments that it generated), or will it only run once it is committed into devel?

@feldergast, @calewis has tested the script and verified its operation (with some minor corner case issues)

1 - From testing (using my forked repo), it will run when the PR is created and after each commit to the PR's related source branch.  However there may be differences between operation that we wont discover till this is merged (My major concern is its operation on PR's coming from forks - which I could not test)

2 - The script affects 3 labels `AT: WIP`, `AT : CLANG-FORMAT PASS`, `AT: CLANG-FORMAT FAIL`.  Only the AT: WIP affects the Autotester the others are just information labels.  If the script fails, it will add a new status (like the pre-merge and pre-test inspections) indicator to the PR that will prevent merging.  This is done by GHA not the Autotester.  Passing will set the status indicator to success and will allow merging.
  * 2.1 - The Autotester and nightly runs will still continue to call the Jenkins job that performs the clang-format test using the script in the core.  This GHA script uses an 3rd party action to perform the clang-format test.
  * 2.2 - There are 2 concerns with this script:
     *  If the developer sets the `AT: WIP` label before this script runs, and the script passes, the WIP label will be removed.
     * There is a race condition between this script and the Autotester run.  The majority of the time this script will be run within 30 seconds of the PR create/commit and then set the WIP flag.  However if the stars align, the AT could process the PR just before the script runs (this is unlikely, but possible)

3 - I though it might run also, however since its not officially part of the Github repo yet, that may be why it did not run.

The script is fairly benign, and once merged, we can submit a few test PR's to ensure that it is working correctly.  If we have problems with it, we can temporarily turn of GHA till the issue can be resolved.

**Review by feldergast — APPROVED** [review]
Questions answered.  I'll go ahead and merge for further testing.
